src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -664,6 +672,88 @@ void GZBridge::odometryCallback(const gz::msgs::OdometryWithCovariance &odometry
 	pthread_mutex_unlock(&_node_mutex);
 }
 
+void GZBridge::laserScanCallback(const gz::msgs::LaserScan &scan)
+{
+	static constexpr int SECTOR_SIZE_DEG = 10; // PX4 Collision Prevention only has 36 sectors of 10 degrees each
+
+	double angle_min_deg = scan.angle_min() * 180 / M_PI;
+	double angle_step_deg = scan.angle_step() * 180 / M_PI;
+
+	int samples_per_sector = std::round(SECTOR_SIZE_DEG / angle_step_deg);
+	int number_of_sectors = scan.ranges_size() / samples_per_sector;
+
+	std::vector<double> ds_array(number_of_sectors, UINT16_MAX);
+
+	// Downsample -- take average of samples per sector
+	for (int i = 0; i < number_of_sectors; i++) {
+
+		double sum = 0;
+
+		int samples_used_in_sector = 0;
+
+		for (int j = 0; j < samples_per_sector; j++) {
+
+			double distance = scan.ranges()[i * samples_per_sector + j];
+
+			// inf values mean no object
+			if (isinf(distance)) {
+				continue;
+			}
+
+			sum += distance;
+			samples_used_in_sector++;
+		}
+
+		// If all samples in a sector are inf then it means the sector is clear
+		if (samples_used_in_sector == 0) {
+			ds_array[i] = scan.range_max();
+
+		} else {
+			ds_array[i] = sum / samples_used_in_sector;
+		}
+	}
+
+	// Publish to uORB
+	obstacle_distance_s obs {};
+
+	// Initialize unknown
+	for (auto &i : obs.distances) {
+		i = UINT16_MAX;
+	}
+
+	obs.timestamp = hrt_absolute_time();
+	obs.frame = obstacle_distance_s::MAV_FRAME_BODY_FRD;
+	obs.sensor_type = obstacle_distance_s::MAV_DISTANCE_SENSOR_LASER;
+	obs.min_distance = static_cast<uint16_t>(scan.range_min() * 100.);
+	obs.max_distance = static_cast<uint16_t>(scan.range_max() * 100.);
+	obs.angle_offset = static_cast<float>(-angle_min_deg);
```

Comment:
Okay I think this large angle offset is the reason for the right side lacking coverage, it shifts the entire array counter clockwise so the obstacle map leaves the right side exposed. Recommend changing it to -5. I’ve used -2.5 here on another rotating LiDAR: https://github.com/PX4/PX4-Autopilot/blob/6a34b63b60251aa15703a66ad9f4479943868f16/src/drivers/distance_sensor/lightware_sf45_serial/lightware_sf45_serial.cpp#L78C11-L78C11

Reply:
Ahh the issue is that this value is already negative

so it should be
```suggestion
	obs.angle_offset = static_cast<float>(angle_min_deg);
```
this gives me the correct internal map array representation 
```
TOPIC: obstacle_distance_fused
 obstacle_distance_fused
    timestamp: 101680000 (0.040000 seconds ago)
    increment: 10.00000
    angle_offset: 0.00000
    distances: [149, 156, 169, 184, 221, 296, 378, 412, 404, 409, 425, 460, 517, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 470, 438, 428, 424, 437, 405, 287, 228, 190, 166, 152, 145, 151, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535]
    min_distance: 10
    max_distance: 3000
    frame: 12
    sensor_type: 0

```
Although this now (correctly?) produces the 90 degree issue in QGC :sweat_smile: So I think QGC needs a fix.

Reply:
i think you are correct, the minus is wrong, if i have the drone orientation as follows:
![image](https://github.com/user-attachments/assets/457b220e-7730-4e59-9970-dbf771e3ac1c)
i get following obstacle_distance
```
TOPIC: obstacle_distance
 obstacle_distance
    timestamp: 221920000 (0.028000 seconds ago)
    increment: 10.00000
    angle_offset: -135.00003
    distances: [695, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 3001, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535, 65535]
    min_distance: 10
    max_distance: 3000
    frame: 12
    sensor_type: 0
 ```
 which would be consisted, as [ObstacleDistance](https://docs.px4.io/main/en/msg_docs/ObstacleDistance.html) states that positive values of angle_offset are to the right. so a negative value for angle_offset corresponds to the setup seen above.